coding/serdes_json.hpp

```diff
@@ -156,7 +165,7 @@ class DeserializerJson
   template <typename T>
   void operator()(std::vector<T> & vs, char const * name = nullptr)
   {
-    json_t * context = SaveContext(name);
+    json_t * outerContext = SaveContext(name);
```

Comment:
Shouldn't you restore context (by using MY_SCOPE_GUARD, for example) if an exception is thrown?